spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/hystrix/JavanicaITest.java

```diff
@@ -0,0 +1,125 @@
+package org.springframework.cloud.sleuth.instrument.hystrix;
+
+import static org.assertj.core.api.BDDAssertions.then;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.netflix.hystrix.EnableHystrix;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.TraceManager;
+import org.springframework.cloud.sleuth.instrument.DefaultTestAutoConfiguration;
+import org.springframework.cloud.sleuth.trace.TraceContextHolder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.jayway.awaitility.Awaitility;
+import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {
+		JavanicaITest.JavanicaITestConfiguration.class })
+public class JavanicaITest {
```

Comment:
I can change the name of the test. `HystrixViaAnnotationsITest` or sth like that.

I don't get the ` inline a Javanica class ` part - can you provide an example? You want me to move the class from the bottom of the file upwards?

I prefer to have the test as such and the crucial fields on top and the technical rest in the bottom part. But of course I can change that.